File: app/src/main/java/com/georgcantor/counter/adapter/HistoryAdapter.java
package com.georgcantor.counter.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.georgcantor.counter.R;
import com.georgcantor.counter.db.History;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;

public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.MyViewHolder> {

    private Context context;
    private List<History> historyList;

    public HistoryAdapter(Context context, List<History> historyList) {
        this.context = context;
        this.historyList = historyList;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.history_list_row, parent, false);
        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        History history = historyList.get(position);
        holder.time.setText(history.getTime());
        holder.dot.setText(Html.fromHtml("&#8226;"));
        holder.timestamp.setText(formatDate(history.getTimestamp()));
    }

    @Override
    public int getItemCount() {
        return historyList.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        public TextView time;
        public TextView dot;
        public TextView timestamp;

        public MyViewHolder(View itemView) {
            super(itemView);
            time = itemView.findViewById(R.id.tv_time);
            dot = itemView.findViewById(R.id.dot);
            timestamp = itemView.findViewById(R.id.timestamp);
        }
    }

    private String formatDate(String dateStr) {
        try {
            SimpleDateFormat fmt = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
            Date date = fmt.parse(dateStr);
            SimpleDateFormat fmtOut = new SimpleDateFormat("MMM d");
            return fmtOut.format(date);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return "";
    }
}

File: app/src/main/java/com/georgcantor/counter/view/adapter/HistoryAdapter.kt
package com.georgcantor.counter.view.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.georgcantor.counter.R
import com.georgcantor.counter.model.Day
import kotlinx.android.synthetic.main.day_item.view.*

class HistoryAdapter(days: MutableList<Day>) :
    RecyclerView.Adapter<HistoryAdapter.HistoryViewHolder>() {

    private val days: MutableList<Day> = ArrayList()

    init {
        this.days.addAll(days)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HistoryViewHolder =
        HistoryViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.day_item, null))

    override fun getItemCount(): Int = days.size

    override fun onBindViewHolder(holder: HistoryViewHolder, position: Int) {
        holder.dateTextView.text = days[position].id
        holder.hoursTextView.text = days[position].hours.toString()
    }

    class HistoryViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val dateTextView: TextView = view.dateTextView
        val hoursTextView: TextView = view.hoursTextView
    }

}